include/filepath.h

```diff
@@ -106,6 +106,8 @@ class Filepath {
 	// Return `true` if Filepath start with `str`, `false` otherwise.
 	bool starts_with(const std::string& str) const;
 
+	std::string file_name() const;
```

Comment:
This should return a `Filepath`, not a `string`, because the return value is a filename. This will also help us avoid all those `unwrap()` calls in the implementation of this method.

Also it should be wrapped in `nonstd::optional` to better match Rust's definition: https://doc.rust-lang.org/nightly/std/path/struct.Path.html#method.file_name

Reply:
https://github.com/newsboat/newsboat/pull/2561/commits/58abe2f58b35ac722a485359ecee475b39608078

Rust type `Option<T>` is unsupported by `cxxbridge`, so I returned usual `String` and proceed empty string as `None`/`nullopt`

Reply:
`String` uses UTF-8, so we can't use it for filenames. Please make it return `Vec<u8>` instead.

I'd also advise to use a separate `bool` out parameter to indicate `None`, so that we could return `Some("")`, but looking at Rust docs, apparently `file_name` returns either `None` or a non-empty string. So I think it's okay to rely on an empty vector in this case.

Reply:
https://github.com/newsboat/newsboat/pull/2561/commits/6338607602742ecf350661d14ae7b6d3d7dc62cf

Reply:
Sorry for changing my mind so often, but since `Vec<u8>` doesn't save us from all those `unwrap()` calls, how about returning `Filepath` from the Rust side? It can be constructed from `OsStr` with `Path::new(str).to_owned()` (I think).

Reply:
We can't use `Path`/`Box<Path>` as return type from Rust side, because it's fixed size array (here is inner implementation):
```rust
pub struct Slice {
    pub inner: [u8],
}
```

which leads to compile-time error:
```
error[E0277]: the size for values of type `[u8]` cannot be known at compilation time
  --> rust/libnewsboat-ffi/src/filepath.rs:16:14
   |
16 |         type Path;
   |              ^^^^ doesn't have a size known at compile-time
   |
   = help: within `filepath::Path`, the trait `Sized` is not implemented for `[u8]`
```
I use `PathBuf` instead.

Reply:
Using a `PathBuf` is the right call!

I think we can fix old compilers quite easily, see my comment below. If that won't work, sure, let's back out this change and write an issue so it can be fixed later.